### PR TITLE
update calculated properies in patient

### LINKED ISSUE #216: Calculate Age and Age Units
calculate age and age units using the diagnosis date

### PR DESCRIPTION
**Pull request by mingjiecn**
changed age, age_unit, age_range and diagnosis_date all to calculated properties. Close #216

### CODE REVIEW

**Review by vsmalladi — APPROVED** [review]
Looks good to me.

**Review by vsmalladi — CHANGES_REQUESTED** [review]
Issue we have is what do we want to show for patients that are 90+ since they will acutally know their age?

File: src/encoded/static/components/search.js [hunk 1]
```diff
@@ -530,8 +530,8 @@ globals.listingViews.register(Target, 'Target');
 class PatientComponent extends React.Component {
     render() {
         const result = this.props.context;
-        const age = (result.age && result.age !== 'unknown') ? ` ${result.age}` : '';
-        const ageUnits = (result.age_units && result.age_units !== 'unknown' && age) ? ` ${result.age_units}` : '';
+        const age = (result.diagnosis.age && result.diagnosis.age !== 'Unknown') ? ` ${result.diagnosis.age}` : '';
+        const ageUnits = (result.diagnosis.age_unit && result.diagnosis.age_unit !== 'Unknown' && age) ? ` ${result.diagnosis.age_unit}` : '';
 
         return (
```
Comment (SenseCode) [inline src/encoded/static/components/search.js]: image

File: src/encoded/types/patient.py [hunk 2]
```diff
@@ -460,11 +438,43 @@ def surgery_summary(self, request, surgery=None):
...
+            "diagnosis_date": {
+                "title": "Diagnosis Date",
+                "description": "Date of Diagnosis",
+                "type": "string",
+            },
+            "age": {
+                "title": "Diagnosis age",
+                "description": "The age of diagnosis.",
+                "type": "string",
+                "pattern": "^((\\d+(\\.[1-9])?(\\-\\d+(\\.[1-9])?)?)|(unknown)|([1-8]?\\d)|(90 or above))$"
```
Comment (SenseCode) [inline src/encoded/types/patient.py]: What does this pattern mean? ((\\d+(\\.[1-9])?(\\-\\d+(\\.[1-9])?)?)|([1-8]?\\d)

File: src/encoded/types/patient.py [hunk 3]
```diff
@@ -496,7 +506,35 @@ def diagnosis_date(self, request, surgery, radiation, medication):
...
+            birth_date = datetime.strptime("1800-01-01", "%Y-%m-%d")
+            end_date = datetime.strptime(diagnosis_date, "%Y-%m-%d")
+            age = end_date.year - birth_date.year -  ((end_date.month, end_date.day) < (birth_date.month, birth_date.day))
+            ageString = str(age) 
+            if age >= 90:
+                ageString = "90 or above"
+
+
+            if age >= 80:
+                age_range = "80+"
```
Comment (SenseCode) [inline src/encoded/types/patient.py:520]: Should we remove age>=90 and add a range for 80-90? So >=90 won't show in our website.

File: src/encoded/static/components/search.js (hunk 1 above)
Comment (vsmalladi) [reply src/encoded/static/components/search.js]: Add unknown to visualization

File: src/encoded/types/patient.py (hunk 2 above)
Comment (vsmalladi) [reply src/encoded/types/patient.py]: Simplify the regular expression to be from and only have units has year. 
0-89
unknown
90 or above

File: src/encoded/types/patient.py (hunk 3 above)
Comment (vsmalladi) [reply src/encoded/types/patient.py:520]: Hold off on this change until I update the years.

File: src/encoded/types/patient.py (hunk 2 above)
Comment (mingjiecn) [reply src/encoded/types/patient.py]: Done

File: src/encoded/static/components/search.js (hunk 1 above)
Comment (mingjiecn) [reply src/encoded/static/components/search.js]: I actually hide the brackets when age is unknown. I feel that unknown may cause confusion.

Comment (vsmalladi) [reply src/encoded/static/components/search.js]: oh i like that better

**Review by vsmalladi — APPROVED** [review]

**Pull request merged**